go.mod

```diff
@@ -1,8 +1,8 @@
 module cloud.google.com/go/alloydbconn
 
-go 1.22.7
+go 1.23
 
-toolchain go1.22.12
+toolchain go1.23.6
```

Comment:
We can leave off the toolchain directive. Verified `go mod tidy` in Go 1.23 and Go 1.24 doesn't add it back.

cc @jackwotherspoon where this might have slipped into Cloud SQL's Go Connector.

Reply:
Oh nice, I've removed it